driver-async/src/test/functional/com/mongodb/async/client/MongoClientsSpecification.groovy

```diff
@@ -118,4 +126,71 @@ class MongoClientsSpecification extends FunctionalSpecification {
         cleanup:
         client?.close()
     }
+
+    @Unroll
+    def 'should correct parse read preference for #readPreference'() {
+        when:
+        def client = MongoClients.create(uri)
+
+        then:
+        client.settings.getReadPreference() == readPreference
+
+        cleanup:
+        client?.close()
+
+        where:
+        uri                                                              | readPreference
+        'mongodb://localhost/?readPreference=secondaryPreferred'         | secondaryPreferred()
```

Comment:
All these different read preferences is more of a test of the ConnectionString class than of the MongoClients class.  It's sufficient to test just one.